src/core/main.c

```diff
@@ -1967,20 +1967,33 @@ static int do_queue_default_job(
                 const char **ret_error_message) {
 
         _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        const char* default_unit;
         Job *default_unit_job;
         Unit *target = NULL;
         int r;
 
-        log_debug("Activating default unit: %s", arg_default_unit);
+        if (arg_default_unit)
+                default_unit = arg_default_unit;
+        else if (in_initrd())
+                default_unit = SPECIAL_INITRD_TARGET;
+        else
+                default_unit = SPECIAL_DEFAULT_TARGET;
+
+        log_debug("Activating default unit: %s", default_unit);
 
-        r = manager_load_startable_unit_or_warn(m, arg_default_unit, NULL, &target);
+        r = manager_load_startable_unit_or_warn(m, default_unit, NULL, &target);
+        if (r < 0 && in_initrd()) {
+                log_info("Falling back to " SPECIAL_INITRD_TARGET ".");
```

Comment:
hmm, we'd typically try initrd.target twice with this, no? doesn't look right?

Reply:
Argh, I made the fall back to the wrong target. Should be fixed now.